### PR TITLE
docs(fetchReleaseNotes): explain platform support, link to overview issue

### LINKED ISSUE #14971: Explain supported platforms for release note fetching
### Describe the proposed change(s).

A user recently asked about fetching release notes when using the Bitbucket Cloud platform and I could not find out from reading our docs. Turns out we do not have a Bitbucket datasource that can fetch the release notes, see issue #14964 and #14138 for more on that.

Our docs should explain on which _platforms_ we support fetching release notes.

Maybe we can list the supported platforms in the [`fetchReleaseNotes` config option](https://docs.renovatebot.com/configuration-options/#fetchreleasenotes) section?

### PR DESCRIPTION
**Pull request by HonkingGoose**
## Changes

- Lists supported platforms
- Add `note` admonition to explain that we can only fetch release notes on some platforms and for some package managers
  - Link to our overview issue

## Context

Closes #14971.

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: docs/usage/configuration-options.md [hunk 1]
```diff
@@ -742,6 +742,21 @@ A similar one could strip leading `v` prefixes:
...
 Set this to `false` if you want to disable release notes fetching.
 
+Renovate can fetch release notes when you're running it on one of these platforms:
+
+- GitHub (.com and Enterprise)
+- GitLab (.com and CE/EE)
+- Bitbucket Cloud
+- Bitbucket Server
+- Azure DevOps
+- Gitea
```
Comment (HonkingGoose) [inline docs/usage/configuration-options.md:752]: I've listed all platforms that Renovate can run on, but we cannot fetch release notes on all platforms.

Can you maintainers please check the code and tell me which platforms we can grab release notes from? 😉

Comment (viceice) [reply docs/usage/configuration-options.md:752]: only on gitlab and GitHub

Comment (HonkingGoose) [reply docs/usage/configuration-options.md:752]:
```suggestion
- GitHub (.com and Enterprise)
- GitLab (.com and CE/EE)
```

File: docs/usage/configuration-options.md [hunk 2]
```diff
@@ -742,6 +742,17 @@ A similar one could strip leading `v` prefixes:
...
+Renovate can fetch release notes when you're running it on one of these platforms:
+
+- GitHub (.com and Enterprise)
+- GitLab (.com and CE/EE)
+
+<!-- prettier-ignore -->
+!!! note
+    Renovate can only show release notes from some platforms and some package managers.
+    We're planning improvements so that Renovate can show more release notes.
+    Read [issue # #14138 on GitHub](https://github.com/renovatebot/renovate/issues/14138) to get a overview of the planned work.
```
Comment (rarkins) [inline docs/usage/configuration-options.md:754]:
```suggestion
    Read [issue 14138 on GitHub](https://github.com/renovatebot/renovate/issues/14138) to get a overview of the planned work.
```

File: docs/usage/configuration-options.md [hunk 3]
```diff
@@ -742,6 +742,17 @@ A similar one could strip leading `v` prefixes:
 
 Set this to `false` if you want to disable release notes fetching.
 
+Renovate can fetch release notes when you're running it on one of these platforms:
+
+- GitHub (.com and Enterprise)
+- GitLab (.com and CE/EE)
```
Comment (astellingwerf) [inline docs/usage/configuration-options.md:748]: With this write up, it remains unclear what I need to do to e.g. my docker images to get Renovate to look up their release notes.

Comment (rarkins) [reply docs/usage/configuration-options.md:748]: I don't think that's relevant for this particular section "fetchReleaseNotes true/false"

File: docs/usage/configuration-options.md [hunk 4]
```diff
@@ -742,6 +742,17 @@ A similar one could strip leading `v` prefixes:
 
 Set this to `false` if you want to disable release notes fetching.
 
+Renovate can fetch release notes when you're running it on one of these platforms:
```
Comment (rarkins) [inline docs/usage/configuration-options.md:745]:
```suggestion
Renovate can fetch release notes when they are hosted on one of these platforms:
```

**Review by rarkins — CHANGES_REQUESTED** [review]

File: docs/usage/configuration-options.md (hunk 3 above)
Comment (HonkingGoose) [reply docs/usage/configuration-options.md:748]: > With this write up, it remains unclear what I need to do to e.g. my docker images to get Renovate to look up their release notes.

@astellingwerf I'm not sure what you mean here. Do you mean:

1. I don't know how to get Renovate to show Docker image release notes.
2. I don't know how to get Renovate to look up any release notes when I'm running Renovate with Docker.

**Review by rarkins — APPROVED** [review]

File: docs/usage/configuration-options.md (hunk 3 above)
Comment (HonkingGoose) [reply docs/usage/configuration-options.md:748]: I've created issue #15052 to discuss:

> With this write up, it remains unclear what I need to do to e.g. my docker images to get Renovate to look up their release notes.

I'm resolving this conversation, so the PR can get merged. 😉

**Review by rarkins — APPROVED** [review]

**Pull request merged**